src/Uno.UI.RemoteControl.Host/RemoteControlServer.cs

```diff
@@ -56,9 +83,26 @@ public RemoteControlServer(IConfiguration configuration)
 		string IRemoteControlServer.GetServerConfiguration(string key)
 			=> _configuration[key] ?? "";
 
-		private void RegisterProcessor(IServerProcessor hotReloadProcessor)
+		private void RegisterProcessor(IServerProcessor processor)
 		{
-			_processors[hotReloadProcessor.Scope] = hotReloadProcessor;
+			if (SharedProcessorPool.Instance.Processors.ContainsKey(processor.Scope))
+			{
+				// Remove any exising processor instances from the same source
+				foreach (var registeredProcessor in SharedProcessorPool.Instance.Processors[processor.Scope])
+				{
+					if (registeredProcessor.GetType().FullName == processor.GetType().FullName)
```

Comment:
Is it possible for this type to not be the actual same type? If it is, we should add a comment. If it's not possible, just compare the actual types

Reply:
Because of the way they're created (via reflection), they _may_ be treated as different types. This seemed the simplest way to determine if we can treat them as the same.